### PR TITLE
allow blank case tags

### PR DESCRIPTION
**Pull request by nickpell**
https://manage.dimagi.com/default.asp?282072

This is a quick, hacky way to unblock the L10K team.  The final approach should require some more thought.

### CODE REVIEW

**Review by esoergel — COMMENTED** [review]
It looks like there are only two one-line changes in the JS - could you instead just reference the feature flag there using
```javascript
hqImport('hqwebapp/js/toggles').toggleEnabled('ALLOW_BLANK_CASE_TAGS')
```
As is, this'll be a "spot-the-differences" nightmare if anyone needs to update these files.

It'd help reviewing if you can commit the copy separately from the changes - I had to set up the diff myself locally to see what the actual changes are.

**Comment by nickpell** [comment]
@esoergel Updated - should be easier to read now.

Thanks for explaining how to reference toggles in JS.

File: corehq/apps/app_manager/static/app_manager/js/forms/advanced/actions.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ hqDefine('app_manager/js/forms/advanced/actions', function() {
             }
             if (!case_type) {
                 return gettext("Case Type required");
-            } else if (!case_tag || self.warn_blank_case_tag()) {
+            } else if (!case_tag || (self.warn_blank_case_tag() && hqImport('hqwebapp/js/toggles').toggleEnabled('ALLOW_BLANK_CASE_TAGS'))) {
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/app_manager/static/app_manager/js/forms/advanced/actions.js]: Shouldn't this be "not toggle enabled"?

File: corehq/apps/app_manager/static/app_manager/js/forms/advanced/actions.js [hunk 2]
```diff
@@ -63,7 +63,9 @@ hqDefine('app_manager/js/forms/advanced/actions', function() {
                 if (!self.auto_select && !tag) {
                     // Don't allow user to blank out case tag
                     self.warn_blank_case_tag(true);
-                    self.case_tag(self.case_tag.previous());
+                    if(hqImport('hqwebapp/js/toggles').toggleEnabled('ALLOW_BLANK_CASE_TAGS')) {
```
Comment (esoergel) [inline corehq/apps/app_manager/static/app_manager/js/forms/advanced/actions.js]: I think this is also backwards (assuming the default is to preserve the old behavior).
Also, what does this part do?

**Review by esoergel — CHANGES_REQUESTED** [review]

**Comment by esoergel** [comment]
Nice, thanks for the switch!

**Review by esoergel — APPROVED** [review]

**Comment by nickpell** [comment]
Merging, since this shouldn't affect tests and L10K is waiting for this

**Comment by orangejenny** [comment]
@nickpell Hmm, this is a tricky issue. L10K may get themselves into the same bind that Jordan ran into. I'll think about potential longer-term solutions - but this is a decent quick fix, thank you! (And thank you for being conscientious and putting it behind a feature flag.)

**Pull request merged**